record-and-playback/presentation/playback/presentation/2.0/playback.js

```diff
@@ -289,235 +281,258 @@ generateThumbnails = function() {
+        }
 
-        //set the current time will fire a "canplaythrough" event to tell us that the video can be played...
-        secondaryMedias[i].currentTime(primaryMedia.currentTime());
-     }
+        if (allMediasAreReady) {
+          mediaSyncing = false;
+          mediaSeeked = false;
+          // Play the master video, causing to play all videos...
+          logger.debug("==Resuming");
+          primaryMedia.play();
+        }
+      }
+    });
   }
-}
+};
+
+function sync() {
+  for (var i = 0; i < secondaryMedias.length ; i++) {
+    if (secondaryMedias[i].media.readyState > 1) {
+      secondaryMedias[i].pause();
+      // Set the current time will fire a "canplay" event to tell us that the video can be played...
+      secondaryMedias[i].currentTime(primaryMedia.currentTime());
+    }
+  }
+};
+
+function setMediaListeners() {
+  // Solo media
+  primaryMedia = Popcorn("#video");
+  primaryMedia.on("seeking", function() {
+    if (primaryMedia.played().length != 0) {
+      mediaSeeked = true;
+    }
+  });
+
+  // When finished seeking
+  primaryMedia.on("seeked", function() {
+    if (!primaryMedia.paused()) {
+      primaryMedia.pause();
+    }
+  });
+
+  primaryMedia.on("canplay", function() {
+    if (mediaSeeked) {
+      mediaSeeked = false;
+      logger.debug("==Resuming");
+      primaryMedia.play();
+    }
+  });
+};
 
-load_script = function(file){
-  console.log("==Loading script "+ file)
-  script = document.createElement('script');
-  script.src = file;
-  script.type = 'text/javascript';
-  document.getElementsByTagName('body').item(0).appendChild(script);
+// Hack for mobile devices that not load media unless they are visible
+function forceMediaEvents() {
+  // When the medias were loaded
+  if (mediaReady) return;
+  if (hasVideo) {
+    logger.debug("==Forcing video ready event");
+    document.dispatchEvent(new CustomEvent('media-ready', {'detail': 'video'}));
+  } else {
+    logger.debug("==Forcing audio ready event");
+    document.dispatchEvent(new CustomEvent('media-ready', {'detail': 'audio'}));
+  }
+  if (hasDeskshare) {
+    logger.debug("==Forcing deskshare ready event");
+    document.dispatchEvent(new CustomEvent('media-ready', {'detail': 'deskshare'}));
+  }
 }
 
 document.addEventListener("DOMContentLoaded", function() {
-  console.log("==DOM content loaded");
+  logger.info("==DOM content loaded");
+  loadMetadata();
+  checkMedias();
+  document.dispatchEvent(new CustomEvent('content-ready', {'detail': 'dom'}));
+}, false);
+
+function loadPlayback() {
+  logger.info("==Loading playback");
   var appName = navigator.appName;
   var appVersion = navigator.appVersion;
 
-  startLoadingBar();
+  // Hack to force mobile devices to show the playback when media do not load while hidden
+  var isMobile = mobileAndTabletCheck();
+  if (isMobile) {
+    logger.info("==Device is mobile");
+    setTimeout(forceMediaEvents, mobileTimeout);
```

Comment:
I don't really like having a timeout here. Does this mean every mobile user has to wait [10 seconds](https://github.com/fcecagno/bigbluebutton/blob/dc0fbc9bd13e26a2895d5f4f3742501fbaa3248a/record-and-playback/presentation/playback/presentation/2.0/lib/writing.js#L40) if the video is hidden by default? That's a long time.

Either we're ok with the media not being loaded (in which case there's no reason to wait), or we should find a way to make the media actually load.

Reply:
The problem is that we don't know if the media is loading and taking a while because the network is slow or it's not loading because the video is hidden. If we find a solution for this, I agree there's no reason to wait.
I suggest we create an issue to revisit this in the future.